Added header and dropdown component

Hornshade opened the pull request:
I created the header and dropdown components. The dropdown component is used for the categories filter in the header. At this point in time it's not filtering.

BiancaDianaAlexiuc commented on src/app/components/dropdown/dropdown.component.scss:
```diff
@@ -0,0 +1,13 @@
+    color: $color-primary-500;
+    font-size: $font-button;
+    font-weight: $font-semibold;
+    ::ng-deep .mat-select-placeholder {
```
Avoid using  ::ng-deep since it's deprecated. Consider adding an HTML class and then style it. If the style from your custom class won't be rendered in the browser move the declaration into styles.scss

BiancaDianaAlexiuc reviewed (commented):
Great use of the BEM methodology. Nice job!

Hornshade commented:
I added the card component, the filters and icons for the categories page. While doing so I created a cards component to view the components .
I fixed the ::ng-deep problem by using a styles.scss declaration.
I fixed some things in the dropdown component aswell like the aligning of the text and outline for the filters.
I have also made a mistake by creating the cards page where I more or less combined 2-3 components which should have been commited and pull requested separately.By doing this I also completed half of another task(the Category Page task #53452). I will try to be more carefull and make commits and PR's more often.

BiancaDianaAlexiuc commented on src/app/components/card/card.component.html:
```diff
@@ -0,0 +1,15 @@
+    <img  class="c-card__image" src={{listing.images[0]}} alt="Card photo">
+    <mat-card-content class="c-card__content">
+      <h1 class="c-card__content--title">{{listing.title}}</h1>
+      <!-- <h4 class="c-card__content--description">{{listing.description}}</h4> -->
```
Don't forget to remove unnecessary comments if you don't need some pieces of code

BiancaDianaAlexiuc commented on src/app/components/card/card.component.html:
```diff
@@ -0,0 +1,15 @@
+    <mat-card-actions class="c-card__actions">
+        <button mat-icon-button class="c-card__actions--button" (click)="setFavorite()">
+            <mat-icon *ngIf="favorite === false">favorite_border</mat-icon>
+            <mat-icon *ngIf="favorite === true" style="color: red;">favorite</mat-icon>
```
To avoid writing duplicate code I suggest using a ternary conditional operator. 
This two lines 
` <mat-icon *ngIf="favorite === false">favorite_border</mat-icon>
   <mat-icon *ngIf="favorite === true" style="color: red;">favorite</mat-icon>`
   
   Could be written like this: 
   
  ` <mat-icon  [ngStyle]="{color: favorite? 'red' : 'white'}">  {{  favorite ? 'favorite ' : 'favorite_border'  }} </mat-icon>`

BiancaDianaAlexiuc commented on src/app/components/cards/cards.component.scss:
```diff
@@ -0,0 +1,66 @@
+        display: flex;
+        flex-direction: row;
+        justify-content: center;
+        text-align: center;
```
There is no need for `text-align: center`; if you use `justify-content: center;`

BiancaDianaAlexiuc commented on src/app/components/dropdown/dropdown.component.ts:
```diff
@@ -0,0 +1,44 @@
+import { Component, Input, OnInit } from '@angular/core';
+
+
+interface Options {
```
You can try to extract all the interfaces into a separate file and just export it.

BiancaDianaAlexiuc replied:
You could create a interfaces folder

BiancaDianaAlexiuc commented on src/app/components/modal/favorite-modal/favorite-modal.component.scss:
```diff
@@ -0,0 +1,31 @@
+        button{
+                font-size: $font-button;
+                font-weight: $font-semibold;
+        }
```
Keep a consistent indentation in SCSS files. You can use the Prettier Visual Code extension that will format the code on save.

Hornshade commented:
I fixed the mentioned things, created an interfaces folder where I made files for the dropdown options and the listings . I made a filters component in order to simplify the cards page , and I moved the cards component into pages( it will be the Category page )
I also made some small changes to dropdowns to have radio-buttons and checkboxes. The header was not fixed when scrolling so i fixed that aswell and i changed the scroll css globaly to fit the theme of the website.
